Add skip/continue functionality to onboarding navigation

vaidehijoshi opened the pull request:
What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Updates the "next" button in the onboarding slides in the three "interactive" components: `FollowTags`, `ProfileForm`, `FollowUsers`. 

For these three slides, the text reads "Skip for now" unless the user has taken an action on the page. If they have (for example, selected a tag, or filled out a field on the form), the button text will update to read "Continue". The button state should return back to "Skip for now" if the user reverts their action (for example, hasn't selected a tag or removes all the text from the form inputs).

Also refactors the `ProfileForm` component to nest values on the `state` within a `formValues` object in order to make it easier to determine whether any of the form inputs have a value or not.

## Related Tickets & Documents
Closes https://github.com/thepracticaldev/dev.to/issues/6544.

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
To QA this feature, you'll want to pull down this branch and check the `FollowTags` and `FollowUsers` components, and make sure that the button text updates correctly:
<img width="823" alt="Screen_Shot_2020-04-23_at_11_28_07_AM" src="https://user-images.githubusercontent.com/6921610/80136320-6536e700-8556-11ea-916b-74a017476380.png">
<img width="826" alt="Screen_Shot_2020-04-23_at_11_28_19_AM" src="https://user-images.githubusercontent.com/6921610/80136365-708a1280-8556-11ea-8e6f-4ceafdd5f088.png">

For the `ProfileForm` component, you'll want to check that the button text updates whenever _any_ input in the form is filled out. **Keep in mind that you need to click off the input, since we're not updating the text on each keystroke, just when the input's value changes.**
<img width="826" alt="Screen Shot 2020-04-23 at 11 28 51 AM" src="https://user-images.githubusercontent.com/6921610/80136379-75e75d00-8556-11ea-88ae-c2bca12493a1.png">
<img width="835" alt="Screen Shot 2020-04-23 at 11 28 42 AM" src="https://user-images.githubusercontent.com/6921610/80136387-7849b700-8556-11ea-9a70-0e60c9ee84fa.png">

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed **(but I added inline docs!)**

vaidehijoshi commented on app/javascript/onboarding/components/ProfileForm.jsx:
```diff
@@ -45,21 +49,31 @@ class ProfileForm extends Component {
-    });
+    currentFormState[name] = value;
+
+    // Once we've derived the new form values, check if the form is empty
+    // and use that value to set the `canSkip` property on the state.
+    const formIsEmpty =
+      Object.values(currentFormState).filter((v) => v.length > 0).length === 0;
```
I don't love this, but it works! I am open to other ways to write this :)

maestromac replied:
I like it 😆 . I can't think of anything better

vaidehijoshi commented:
Included one additional feature in this PR since it was easy enough to do! The final onboarding slide now has special button text, which reads "Finish". You can see this reflected in the last commit. This was something we decided on with @lisasy's input (and isn't reflected in the design):

<img width="826" alt="Screen Shot 2020-04-23 at 12 02 37 PM" src="https://user-images.githubusercontent.com/6921610/80138996-96b1b180-855a-11ea-9aff-793fc3bd008d.png">

vaidehijoshi commented:
While waiting for review on this, I noticed that the back button was growing in size/width because of changes introduced in this PR. I have pushed up a commit to fix this, and I've also addressed [an accessibility concern](https://github.com/thepracticaldev/dev.to/pull/7455#discussion_r413913391) that @lisasy brought up in a recent PR!

vaidehijoshi commented:
I noticed an annoying thing and didn't want to merge/deploy this until I fixed it. When I previously fixed my arrow styles, I caused a regression that misaligned the stepper such that it was no longer centered: 
<img width="828" alt="Screen_Shot_2020-04-23_at_3_13_05_PM" src="https://user-images.githubusercontent.com/6921610/80154931-662b4100-8575-11ea-927f-bdde4c48f6ed.png">

I fixed this regression and amended my previous commit, so that everything is centered as per the designs :)
<img width="828" alt="Screen_Shot_2020-04-23_at_3_14_17_PM" src="https://user-images.githubusercontent.com/6921610/80154924-64617d80-8575-11ea-9590-33bd6cdbbed4.png">

lisasy commented on app/javascript/onboarding/__tests__/__snapshots__/Onboarding.test.jsx.snap:
```diff
@@ -7,21 +7,23 @@ preact-render-spy (1 nodes)
-          </svg>
-        </button>
+            <svg
+              width="24"
```
@vaidehijoshi Yay! 😀

lisasy commented on app/assets/stylesheets/preact/onboarding-modal.scss:
```diff
@@ -126,18 +126,41 @@
+      $button-size: 40px; // Explicitly set size for accessibility.
+
+      border-radius: 100%;
+      width: $button-size;
```
I think that instead of width which is a fixed value, setting this as `min-width: $button-size` may ensure that button is always a minimum size.

vaidehijoshi replied:
Ah, I wasn't really thinking that the width was ever going to change/grow for this button, since it has an explicitly-sized `svg` within it, which is why I set it as `width` directly 🤷. But I've gone ahead and changed it to `min-width`.

lisasy commented:
One small comment on using `min-width` instead of `width `to set the button size, but overall, it LGTM!
